pyedflib/edfwriter.py

```diff
@@ -292,7 +377,17 @@ def update_header(self) -> None:
         # the channels, we need to find a common denominator so that all sample
         # frequencies can be represented accurately.
         # this can be overwritten by explicitly calling setDatarecordDuration
-        self._calculate_optimal_record_duration()
+
+        for ch in self.channels:
+            # raise exception, can be removed in later release
+            if 'sample_rate' in ch:
+                raise ValueError('Use of `sample_rate` is deprecated, use `sample_frequency` instead')
```

Comment:
Are you sure you don't want to raise a `FutureWarning` instead?

Reply:
right! that makes sense :)